server/redeemer.go

```diff
@@ -0,0 +1,383 @@
+	t := pmTicket(ticket)
+	if r.recipient != t.Recipient {
+		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("invalid ticket recipient 0x%x, expected %v", ticket.TicketParams.Recipient, r.recipient.Hex()))
+	}
+	if err := r.sm.QueueTicket(t); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	glog.Infof("ticket queued sender=0x%x", ticket.Sender)
+
+	return &net.QueueTicketRes{}, nil
+}
+
+// MaxFloat is a unary RPC method to request the max float value for a sender
+func (r *Redeemer) MaxFloat(ctx context.Context, req *net.MaxFloatReq) (*net.MaxFloatUpdate, error) {
+	maxFloat, err := r.sm.MaxFloat(ethcommon.BytesToAddress(req.Sender))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &net.MaxFloatUpdate{MaxFloat: maxFloat.Bytes()}, nil
+}
+
+// MonitorMaxFloat starts a server-side stream to the client to send max float updates for sender specified in the request
+func (r *Redeemer) MonitorMaxFloat(req *net.MaxFloatReq, stream net.TicketRedeemer_MonitorMaxFloatServer) error {
+	sender := ethcommon.BytesToAddress(req.Sender)
+
+	sink := make(chan struct{}, 10)
+	sub := r.sm.SubscribeMaxFloatChange(sender, sink)
+	defer sub.Unsubscribe()
+	for {
+		select {
+		case <-r.quit:
+			return nil
+		case err := <-sub.Err():
+			if err == nil {
+				return status.Error(codes.Canceled, "subscription closed")
+			}
+			return status.Error(codes.Canceled, err.Error())
+		case <-sink:
+			maxFloat, err := r.sm.MaxFloat(sender)
+			if err != nil {
+				glog.Error(err)
+			}
+			if err := stream.Send(&net.MaxFloatUpdate{MaxFloat: maxFloat.Bytes()}); err != nil {
+				glog.Error(err)
+				if err == io.EOF {
+					return err
+				}
+			}
+		}
+	}
+}
+
+type RedeemerClient struct {
+	conn *grpc.ClientConn
+	rpc  net.TicketRedeemerClient
+
+	senders map[ethcommon.Address]*remoteSender
+	mu      sync.RWMutex
+	quit    chan struct{}
+	sm      pm.SenderManager
+	tm      pm.TimeManager
+}
+
+type remoteSender struct {
+	maxFloat   *big.Int
+	stream     net.TicketRedeemer_MonitorMaxFloatClient
+	done       chan struct{}
+	lastAccess time.Time
+}
+
+// NewRedeemerClient instantiates a new client for the ticket redemption service
+// The client implements the pm.SenderMonitor interface
+func NewRedeemerClient(uri *url.URL, sm pm.SenderManager, tm pm.TimeManager) (*RedeemerClient, error) {
+	conn, err := grpc.Dial(
+		uri.String(),
+		grpc.WithBlock(),
+		grpc.WithTimeout(GRPCConnectTimeout),
+		grpc.WithInsecure(),
+	)
+
+	//TODO: PROVIDE KEEPALIVE SETTINGS
+	if err != nil {
+		glog.Errorf("Did not connect to orch=%v err=%v", uri, err)
+		return nil, fmt.Errorf("Did not connect to orch=%v err=%v", uri, err)
+	}
+	return &RedeemerClient{
+		conn:    conn,
+		rpc:     net.NewTicketRedeemerClient(conn),
+		sm:      sm,
+		tm:      tm,
+		senders: make(map[ethcommon.Address]*remoteSender),
+		quit:    make(chan struct{}),
+	}, nil
+}
+
+func (r *RedeemerClient) Start() {
+	r.startCleanupLoop()
```

Comment:
Needs to be in a goroutine or else this method will block.

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1513/commits/a8e3d49938dcf3a62e11bff72accdc054c2d205d